### ISSUE TITLE
`if: always()` with `cache/save` may lead to errors (fix README!)

### ISSUE THREAD

**Issue by andreasabel** [actions/cache#1154]
https://github.com/actions/cache/blob/04f198bf0b2a39f7230a4304bf07747a0bddf146/save/README.md?plain=1#L83-L84

In my experience, if we had an exact cache hit, the attempt to save the cache (under the same key) will produce an error.
So, the docs should be updated to something like
```
if: always() && !steps.<cache-restore-step>.outputs.cache-hit
```
at least in cases where we did a restore.

As `cache-hit` will only be true if the `cache-primary-key` equals the `cache-matched-key` this will always save the cache when it is possible.

**Comment by andreasabel** [comment]
Issue is still unfixed:
https://github.com/actions/cache/blob/704facf57e6136b1bc63b828d79edcd491f0ee84/save/README.md?plain=1#L83-L84

**Comment by andreasabel** [comment]
ATTN: @bethanyj28 Please triage/fix.

**Comment by andreasabel** [comment]
Rather
```
if: always() && steps.<cache-restore-step>.outputs.cache-hit != 'true'
```

**Comment by joshmgross** [comment]
Fixing in https://github.com/actions/cache/pull/1452

**Comment by joshmgross** [comment]
Fixed in #1452
